### PR TITLE
HMIS viewable entities

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: app/models/concerns/hmis_enabled.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HmisEnabled
+  extend ActiveSupport::Concern
+  included do
+    def self.hmis_enabled?
+      ENV['ENABLE_HMIS_API'] == 'true' && RailsDrivers.loaded.include?(:hmis)
+    end
```
Comment (awisema) [inline app/models/concerns/hmis_enabled.rb]: Could `application_helper.rb` and `enforce_hmis_enabled.rb` use this instead of having 3 copies?

Comment (awisema) [reply app/models/concerns/hmis_enabled.rb]: And maybe pull `hmis_admin_visible?` into here as well?

File: drivers/hmis/app/models/hmis/hud/assessment.rb [hunk 2]
```diff
@@ -33,7 +33,7 @@ class Hmis::Hud::Assessment < Hmis::Hud::Base
   scope :in_progress, -> { where(enrollment_id: WIP_ID) }
 
   scope :viewable_by, ->(user) do
-    enrollment_ids = Hmis::Hud::Enrollment.viewable_by(user).pluck(:id, :EnrollmentID)
+    enrollment_ids = Hmis::Hmis::Enrollment.viewable_by(user).pluck(:id, :EnrollmentID)
```
Comment (awisema) [inline drivers/hmis/app/models/hmis/hud/assessment.rb]: I think these changes (here, enrollment, funder, inventory, and project coc), are accidental?

File: app/views/admin/roles/_new_form.haml [hunk 3]
```diff
@@ -1,8 +1,4 @@
-= simple_form_for [:admin, @role] do |f|
-  = f.error_notification
+= f.error_notification
 
-  .form-inputs
-    = f.input :name, label: 'Role name'
-
-  .form-actions
```
Comment (awisema) [inline app/views/admin/roles/_new_form.haml:7]: Maybe pass `value` into `new_form` so, you can leave this in here?

File: drivers/hmis/app/models/hmis/hud/organization.rb [hunk 4]
```diff
@@ -18,7 +19,8 @@ class Hmis::Hud::Organization < Hmis::Hud::Base
 
   # Any organizations the user has been assigned, limited to the data source the HMIS is connected to
   scope :viewable_by, ->(user) do
-    viewable_ids = GrdaWarehouse::Hud::Organization.viewable_by(user).pluck(:id)
+    viewable_ids += user.organizations..pluck(:id)
```
Comment (awisema) [inline drivers/hmis/app/models/hmis/hud/organization.rb]:
```suggestion
    viewable_ids += user.organizations.pluck(:id)
```

**Review by awisema — CHANGES_REQUESTED** [review]

File: drivers/hmis/app/models/hmis/hud/assessment.rb (hunk 2 above)
Comment (eanders) [reply drivers/hmis/app/models/hmis/hud/assessment.rb]: Oh, yeah, messed those up.

**Comment by gigxz** [comment]
Looks good! I tested it out and confirmed that only assigned projects are viewable in the HMIS 🥳 
~~I don't see a way to assign HMIS Roles to users/groups. Is that missing, or am I not seeing it?~~ nevermind, I found it eventually by clicking on the role name. I didn't think of that.

**Comment by eanders** [comment]
Yeah, the assignment of users to roles and groups is slightly awkward at the moment, but it's there!

**Review by gigxz — APPROVED** [review]
Looks good to me!

Just to confirm what I'm seeing: all _clients_ in the user's HMIS Data source are visible to the user. What this looks like in the HMIS, if you're only assigned to a couple projects: you can still search across clients, but most of the clients will appear as if they have no enrollments. I think that's all correct.

**Comment by eanders** [comment]
That's the current setup, we may want to apply a `visible_to` to Client, but we haven't yet.

**Pull request merged**